### PR TITLE
build with gradle-intellij-plugin

### LINKED ISSUE #175: Need to clear plugin-sandbox when making changes to login plugin
<idea-config>/system/plugins-sandbox doesn't reflect changes to the plugin sandbox when the "Core-IJ-Community" run configuration is run unless it is missing the login dependency.

It's not really clear to me why, perhaps because the login plugin dependency is "provided"?

### PR DESCRIPTION
**Pull request by patflynn**
I've updated how to use this new setup for importing into IntelliJ. Due
to an issue with how intellij does import you unfortunately have to
revert the imports changes to .idea after import to get the
preconfigured runconfiguration. There's also a couple ugly hacks to support
multi-plugin projects which will be fixed when
 https://github.com/JetBrains/gradle-intellij-plugin/issues/17 is fixed
in a few days.

closes #175, #167, #161, #131, #90

### CODE REVIEW

**Comment by patflynn** [comment]
@elharo @nbashirbello @loosebazooka

File: .idea/compiler.xml [hunk 1]
```diff
@@ -18,6 +18,5 @@
         <processorPath useClasspath="true" />
       </profile>
     </annotationProcessing>
-    <bytecodeTargetLevel target="1.6" />
   </component>
```
Comment (loosebazooka) [inline .idea/compiler.xml]: Don't we need to keep this requirement for 1.6 target level?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -41,37 +41,17 @@ Other useful targets while developing include:
...
-
-### Optional:  Downloading source for IDEA to debug.
-
-[Download the IDEA source tarball from JetBrains](http://www.jetbrains.org/display/IJOS/Download) 
-into your home (or any other convenient) directory. Extract it and 
-add the resulting directory (usually something like ideaIC-141.1532.4) 
-in the Sourcepath tab of Project Structure > SDKs for the IntelliJ CE 141.1532 SDK.
-
+1. 'New project from existing sources'
+1. Select the root build.gradle file to import
```
Comment (loosebazooka) [inline CONTRIBUTING.md]: all numbered one? Is this something markdown takes care of automatically?

File: core-plugin/build.gradle [hunk 3]
```diff
@@ -1,64 +1,53 @@
-apply from: '../scripts/javac2.gradle'
-def ideaHome = '../idea-IC/'
-def pluginName = 'google-cloud-tools-core'
+plugins {
+    id "org.jetbrains.intellij" version "0.0.29"
+}
+
+// **replace** default sourceSets with idea layout
```
Comment (loosebazooka) [inline core-plugin/build.gradle]: Do you think we can standardize around the gradle layout eventually? Or is that a bad idea?

File: .idea/compiler.xml (hunk 1 above)
Comment (patflynn) [inline .idea/compiler.xml]: not really sure what's going on here. language level is specified in the gradle file, I'm suprised it's removed here.

File: CONTRIBUTING.md (hunk 2 above)
Comment (patflynn) [inline CONTRIBUTING.md]: yeah, that's a numbered bullet markdown feature.

File: google-account-plugin/build.gradle [hunk 4]
```diff
@@ -1,84 +1,36 @@
-// see https://github.com/JetBrains/gradle-intellij-plugin
+plugins {
+    id "org.jetbrains.intellij" version "0.0.29"
```
Comment (loosebazooka) [inline google-account-plugin/build.gradle]: Are we gonna use 28 or 29 across plugins? Can we do this in the root project and apply it in the subprojects {} closure so we don't have to keep this in 4 different files

File: core-plugin/build.gradle (hunk 3 above)
Comment (patflynn) [inline core-plugin/build.gradle]: I would like to do that. Not in this PR though.

**Comment by patflynn** [comment]
I removed the non-used property from gradle.properties

File: common-test-lib/build.gradle [hunk 5]
```diff
@@ -1,10 +1,9 @@
-apply plugin: 'java'
-
-repositories {
-    mavenCentral()
+// Not an IJ plugin project, but included to add IntelliJ platform dependencies to the classpath
+plugins {
+    id "org.jetbrains.intellij" version "0.0.28"
```
Comment (loosebazooka) [inline common-test-lib/build.gradle]: 0.0.29

File: common-lib/build.gradle [hunk 6]
```diff
@@ -1,27 +1,8 @@
-apply plugin: 'java'
-
-repositories {
-    mavenCentral()
+// Not an IJ plugin project, but included to add IntelliJ platform dependencies to the classpath
+plugins {
+    id "org.jetbrains.intellij" version "0.0.28"
```
Comment (loosebazooka) [inline common-lib/build.gradle]: 0.0.29

File: build.gradle [hunk 7]
```diff
@@ -1,36 +1,44 @@
...
+        classpath 'org.springframework.build.gradle:propdeps-plugin:0.0.7'
+        classpath 'gradle.plugin.org.jetbrains:gradle-intellij-plugin:0.0.29'
+    }
 }
 
-apply plugin: 'base'
-
-task loadIdea(type:Exec) {
-    commandLine './fetchIdea.sh'
+configure(allprojects) {
```
Comment (loosebazooka) [inline build.gradle]: do these have to be applied at the "allprojects" level? or can be at the "subprojects" level?

**Comment by patflynn** [comment]
all fixed up.  propdeps only applied to subprojects.

**Pull request merged**